distance-metrics/src/metric.rs

```diff
@@ -0,0 +1,15 @@
+use super::types::VectorElementType;
+use super::types::ScoreType;
+
+/// Defines how to compare vectors
+pub trait Metric {
+    /// Greater the value - more distant the vectors
+    fn distance(v1: &[VectorElementType], v2: &[VectorElementType]) -> ScoreType;
+
+    /// Necessary vector transformations performed before adding it to the collection (like normalization)
+    /// Return None if metric does not required preprocessing
+    fn preprocess(vector: &[VectorElementType]) -> Option<Vec<VectorElementType>>;
+
+    /// correct metric score for displaying
+    fn postprocess(score: ScoreType) -> ScoreType;
```

Comment:
Why do we need a post-processing step instead of inlining this with the rest of the metric computation?